go/parquet/reader_properties.go

```diff
@@ -0,0 +1,73 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parquet
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/apache/arrow/go/arrow/ipc"
+	"github.com/apache/arrow/go/arrow/memory"
+	"golang.org/x/xerrors"
+)
+
+// ReaderProperties are used to define how the file reader will handle buffering and allocating buffers
+type ReaderProperties struct {
+	alloc memory.Allocator
+	// Default buffer size to utilize when reading chunks
+	BufferSize int64
+	// create with NewFileDecryptionProperties if dealing with an encrypted file
+	FileDecryptProps *FileDecryptionProperties
+	// If this is set to true, then the reader will use SectionReader to
+	// just use the read stream when reading data. Otherwise we will buffer
+	// the data we're going to read into memory first and then read that buffer.
+	BufferedStreamEnabled bool
```

Comment:
it might pay to add a little more commentary here on why one would choose one over the other.

Reply:
done